### PR TITLE
feat(chat): add conversation compaction to prevent context window overflow

### PR DESCRIPTION
**Pull request by hazemadelkhalel**
## Summary
Adds server-side chat compaction that automatically summarizes older conversation messages when the context window is running out of space. This prevents provider 400 errors on long conversations and reduces token costs.

### Problem
The chat feature stores the entire conversation history in a JSONB `messages` column and sends **all messages** to the LLM on every request. When conversations grow beyond the model's context window, the provider returns a 400 error and the conversation breaks with no recovery.

### Solution — Hybrid Compaction (Summary + Recent Window)
Before each `streamText()` call, the server estimates the token count and checks if it exceeds 70% of the provider's context limit. When triggered:

1. **Summarize** — Calls `generateText()` with a structured prompt that preserves user decisions, entity names, tool call results, and the current task
2. **Split** — Keeps recent messages verbatim (the "recent window") and replaces older messages with the summary
3. **Store** — Saves the summary in new `summary` and `summarizedUpToIndex` columns; the full `messages` array is preserved for UI rendering
4. **Send** — Passes `[summary_message, ...recent_messages]` to the LLM instead of the full history

The compaction is **fully transparent** — the user sees the complete message history in the UI, and the model retains context about earlier messages via the summary.

### Key Design Decisions
| Decision | Choice |
|---|---|
| Trigger | Lazy — only at 70% of provider context limit |
| Token estimation | Character heuristic (~4 chars/token) |
| Context limits | Per-provider map (OpenAI: 1M, Anthropic: 200K, Google: 1M, etc.) with 128K default |
| Summarization model | Same model as the conversation |
| Storage | Separate `summary` + `summarizedUpToIndex` columns; `messages` untouched |
| Tool call safety | Respects assistant→tool message pairs to avoid orphaned `tool_result` errors |
| Overflow fallback | Progressively shrinks recent window; throws `CHAT_CONTEXT_LIMIT_EXCEEDED` if even minimal payload is too large |

### New Files
- `packages/server/api/src/app/chat/chat-compaction.ts` — Compaction module (estimateTokenCount, shouldCompact, compactMessages, buildCompactedPayload)
- `packages/server/api/src/app/database/migration/postgres/1786000000000-AddChatCompactionColumns.ts` — Migration adding `summary` and `summarizedUpToIndex` columns
- `packages/server/api/test/unit/app/chat/chat-compaction.test.ts` — 13 unit tests

### Modified Files
- `chat-service.ts` — Integrated compaction into `sendMessage()` via `resolveCompactionState` helper
- `chat-conversation-entity.ts` — Added `summary` (String, nullable) and `summarizedUpToIndex` (Number, nullable) columns
- `ai-providers/index.ts` — Added `PROVIDER_MAX_CONTEXT_TOKENS` map and `aiProviderUtils.getMaxContextTokens()`
- `activepieces-error.ts` — Added `CHAT_CONTEXT_LIMIT_EXCEEDED` error code
- `error-handler.ts` — Mapped new error code to 400
- `chat/index.ts` (shared) — Added fields to `ChatConversation` schema
- `shared/package.json` — Bumped to 0.68.6

## Test plan
- [x] Unit tests for `estimateTokenCount`, `shouldCompact`, `buildCompactedPayload` including tool boundary edge cases (13 tests)
- [x] End-to-end manual test: compaction triggered on real conversation, summary stored in DB, model retained context from summarized messages
- [x] Verified tool call boundary handling prevents Anthropic `tool_result` pairing errors
- [ ] Verify on OpenAI and Google providers
- [ ] Verify re-compaction (extending an existing summary with more messages)
- [x] `npm run lint-dev` passes with 0 errors
- [x] TypeScript compiles cleanly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/server/api/src/app/chat/chat-compaction.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+const COMPACTION_THRESHOLD = 0.7
+const RECENT_WINDOW_RATIO = 0.3
+const CHARS_PER_TOKEN_ESTIMATE = 4
+const MIN_MESSAGES_BEFORE_COMPACTION = 6
+const ESTIMATED_TOKENS_PER_MESSAGE = 200
+
+const COMPACTION_SYSTEM_PROMPT = readFileSync(
+    path.resolve('packages/server/api/src/assets/prompts/chat-compaction-prompt.md'),
+    'utf8',
+)
```
[the remark this answers is not in the document]
Comment (hazemadelkhalel) [reply packages/server/api/src/app/chat/chat-compaction.ts:16]: That's wrong we should take the full path
